File: lua/abstract/plugins/ts-autotag.lua
--[[
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
─────────────────────────────────────────────────
Plugin:    nvim-ts-autotag
Github:    github.com/windwp/nvim-ts-autotag

 Use treesitter to auto close and auto rename html tag
─────────────────────────────────────────────────
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
--]]

local spec = {
	"windwp/nvim-ts-autotag",
	event = "InsertEnter",
	dependencies = {
		"nvim-treesitter/nvim-treesitter",
	},
	-- stylua: ignore
	ft = {
		"astro", "glimmer", "handlebars", "hbs", "html", "javascript",
		"javascriptreact", "jsx", "markdown", "php", "rescript",
		"svelte", "tsx", "typescript", "typescriptreact", "vue", "xml",
	},
}

spec.config = function()
	require("nvim-ts-autotag").setup()
end

return spec

File: code/src/neovim/editor/src/plugins/configs/ts_autotag.rs
/*
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
────────────────────────────────────────────────
Plugin: nvim-ts-autotag
Source: github.com/windwp/nvim-ts-autotag

Use treesitter to auto close and auto rename html tag
────────────────────────────────────────────────
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
*/

use crate::lua_spec;

pub struct Plugin;

impl Plugin {
    pub fn spec() -> crate::plugins::spec::SpecInfo {
        // language=lua
        lua_spec!(raw r#"{
            "windwp/nvim-ts-autotag",
            event = { "InsertEnter", "LspAttach" },
            -- stylua: ignore
            ft = {
                "astro", "glimmer", "handlebars", "hbs", "html", "javascript",
                "javascriptreact", "jsx", "markdown", "php", "rescript",
                "svelte", "tsx", "typescript", "typescriptreact", "vue", "xml",
            },
            opts = {},
        }"#)
    }
}
